### DISCUSSION TITLE
install dependencies

### DISCUSSION THREAD

**Discussion by dmbr0** [bashly-framework/bashly#520] [General]
Is there a way to install unmet dependencies instead of suggesting the user install them?

**Comment by DannyBen** [comment]
I don't think that is the responsibility of the command line tool to install other dependencies.

Do you know of any other tool that does that?
Besides, installing dependencies varies wildly between operating systems, package managers, and personal preference.

Of course, if you wish to do it for your own bashly-generated tool, just avoid using the built in `dependencies` directive, and implement your own logic to install the things you need.

**Reply by dmbr0** [reply]
True true. Guess I was thinking for CI/CD purposes and would be nice to have the yaml syntax.

**Reply by dragonfleas** [reply]
@dmbr0 It's not yaml configured, but look into the Nix package manager, it's a good tool for declarative package configuration. I'm sure there's probably a way for you to have your own initialization hook in a bashly setup that runs a nix config. Specifically look at https://github.com/DavHau/nix-portable for a CI/CD portable solution.

**Reply by dmbr0** [reply]
Thank you sir I will look into it. 🫡
